### PR TITLE
Mainly remove misleading error on node stop

### LINKED ISSUE #228: node: Implement graceful shutdown
Related to #82.

Reference [this article](https://rudderstack.com/blog/implementing-graceful-shutdown-in-go/) for new ideas.

### PR DESCRIPTION
**Pull request by Wondertan**
Closes #228

### CODE REVIEW

File: service/header/subscription.go [hunk 1]
```diff
@@ -30,15 +30,14 @@ func newSubscription(topic *pubsub.Topic) (*subscription, error) {
 func (s *subscription) NextHeader(ctx context.Context) (*ExtendedHeader, error) {
 	msg, err := s.subscription.Next(ctx)
 	if err != nil {
-		log.Errorw("reading next message from subscription", "err", err.Error())
```
Comment (Wondertan) [inline service/header/subscription.go:33]: Note for @renaynay. There is no value in logging the error here as the only types of errors we will see here are context-related. And users of NextHeader should handle them accordingly as we do in DASer.

File: service/header/subscription.go [hunk 2]
```diff
@@ -30,15 +30,14 @@ func newSubscription(topic *pubsub.Topic) (*subscription, error) {
...
 		return nil, err
 	}
 	log.Debugw("received message", "topic", msg.Message.GetTopic(), "sender", msg.ReceivedFrom)
 
 	var header ExtendedHeader
 	err = header.UnmarshalBinary(msg.Data)
 	if err != nil {
-		log.Errorw("unmarshaling data from message", "err", err.Error())
+		log.Errorw("unmarshalling data from message", "err")
```
Comment (adlerjohn) [inline service/header/subscription.go:40]: Is there a reason to remove `err.Error()` from the string? Now the string is just "unmarshalling data from message err" all the time, no?

Comment (renaynay) [reply service/header/subscription.go:40]: Yeah, i don't understand why to remove `unmarshaling` as well as the error message?

Comment (Wondertan) [reply service/header/subscription.go:40]: Oh, ok. Thank you for catching this. I wanted to only remove `.Error()` part.

Comment (Wondertan) [reply service/header/subscription.go]: Fixed

**Review by renaynay — APPROVED** [review]

File: service/header/subscription.go (hunk 2 above)
Comment (Wondertan) [reply service/header/subscription.go]: @adlerjohn, resolved and approved?

**Review by adlerjohn — APPROVED** [review]

**Pull request merged**